src/python_testing/TC_RVCOPSTATE_2_3.py

```diff
@@ -0,0 +1,145 @@
+from matter_testing_support import MatterBaseTest, async_test_body, default_matter_test_main, type_matches
+from mobly import asserts
+
+# This test requires several additional command line arguments
+# run with
+# --int-arg PIXIT_ENDPOINT:<endpoint>
+
+
+class TC_RVCOPSTATE_2_3(MatterBaseTest):
+
+    async def read_mod_attribute_expect_success(self, endpoint, attribute):
+        cluster = Clusters.Objects.RvcOperationalState
+        return await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attribute)
+
+    async def send_pause_cmd(self) -> Clusters.Objects.RvcOperationalState.Commands.Pause:
+        ret = await self.send_single_cmd(cmd=Clusters.Objects.RvcOperationalState.Commands.Pause(), endpoint=self.endpoint)
+        asserts.assert_true(type_matches(ret, Clusters.Objects.RvcOperationalState.Commands.OperationalCommandResponse),
+                            "Unexpected return type for Pause")
+        return ret
+
+    async def send_resume_cmd(self) -> Clusters.Objects.RvcOperationalState.Commands.Resume:
+        ret = await self.send_single_cmd(cmd=Clusters.Objects.RvcOperationalState.Commands.Resume(), endpoint=self.endpoint)
+        asserts.assert_true(type_matches(ret, Clusters.Objects.RvcOperationalState.Commands.OperationalCommandResponse),
+                            "Unexpected return type for Resume")
+        return ret
+
+    @async_test_body
+    async def test_TC_RVCOPSTATE_2_3(self):
+
+        asserts.assert_true('PIXIT_ENDPOINT' in self.matter_test_config.global_test_params,
+                            "PIXIT_ENDPOINT must be included on the command line in "
+                            "the --int-arg flag as PIXIT_ENDPOINT:<endpoint>")
+
+        self.endpoint = self.matter_test_config.global_test_params['PIXIT_ENDPOINT']
+
+        asserts.assert_true(self.check_pics("RVCOPSTATE.S.A0002"), "RVCOPSTATE.S.A0002 must be supported")
+        asserts.assert_true(self.check_pics("RVCOPSTATE.S.A0003"), "RVCOPSTATE.S.A0003 must be supported")
+        asserts.assert_true(self.check_pics("RVCOPSTATE.S.A0004"), "RVCOPSTATE.S.A0004 must be supported")
+        asserts.assert_true(self.check_pics("RVCOPSTATE.S.C00.Rsp"), "RVCOPSTATE.S.C00.Rsp must be supported")
+        asserts.assert_true(self.check_pics("RVCOPSTATE.S.C03.Rsp"), "RVCOPSTATE.S.C03.Rsp must be supported")
+
+        attributes = Clusters.RvcOperationalState.Attributes
+
+        self.print_step(1, "Commissioning, already done")
+
+        self.print_step(2, "Manually put the device in a state where it can receive a Pause command")
+        input("Press Enter when done.\n")
+
+        self.print_step(3, "Read OperationalStateList attribute")
+        op_state_list = await self.read_mod_attribute_expect_success(endpoint=self.endpoint, attribute=attributes.OperationalStateList)
+
+        logging.info("OperationalStateList: %s" % (op_state_list))
+
+        defined_state_ids = {0x01: 'Running',
```

Comment:
For here and in other tests - we have defines of these already from the codegen. If you use those, you won't need to worry about them changing under you or having to update if more get added

Clusters.OperationalState.Enums.OperationalStateEnum

spec question - that enum also includes a kStopped - is that not mandatory?

Reply:
We need that one plus a few others:

0x00 Stopped

0x40 SeekingCharger
0x41  Charging 
0x42 Docked

Reply:
But I don't think this test should check for that, since this is covered by RVC OpState TC 2.1